### PR TITLE
[Merged by Bors] - Log hostname when spacemesh node is started

### LINKED ISSUE #2318: Print listening IP address on startup
Right now, we print logs such as the following when the node starts. We log information on ports but we should also log the interface/IP that the node is listening on. This would help a lot with debugging managed miners (so it's clearer which VM each miner is running on).

```
2021-03-04T13:58:18.886-0500    INFO    9979a.gossipListener    start listening {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3"}
2021-03-04T13:58:18.886-0500    INFO    9979a.gossipListener    start listening {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3"}
...
2021-03-04T13:58:18.886-0500    DEBUG   9979a.p2p               Starting the p2p layer  {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3", "P2PID": "8sh6XhQN8mHeoxGivPLzviSAohGTUfjEW36k5aVwoWbj"}
2021-03-04T13:58:18.886-0500    INFO    9979a.p2p               Trying to acquire ports using UPnP, this might take a while..   {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3", "P2PID": "8sh6XhQN8mHeoxGivPLzviSAohGTUfjEW36k5aVwoWbj"}
...
2021-03-04T13:58:46.790-0500    INFO    9979a.p2p          .tcpnet              Started TCP server listening for connections on tcp:[::]:7153   {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3", "P2PID": "8sh6XhQN8mHeoxGivPLzviSAohGTUfjEW36k5aVwoWbj"}
2021-03-04T13:58:46.790-0500    INFO    9979a.p2p          .udpnet              Started UDP server listening for messages on udp:[::]:7153      {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3", "P2PID": "8sh6XhQN8mHeoxGivPLzviSAohGTUfjEW36k5aVwoWbj"}
2021-03-04T13:58:46.790-0500    DEBUG   9979a.p2p               Starting to listen for network messages {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3", "P2PID": "8sh6XhQN8mHeoxGivPLzviSAohGTUfjEW36k5aVwoWbj"}
2021-03-04T13:58:46.790-0500    DEBUG   9979a.p2p               starting the udp server {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3", "P2PID": "8sh6XhQN8mHeoxGivPLzviSAohGTUfjEW36k5aVwoWbj"}
2021-03-04T13:58:46.790-0500    DEBUG   9979a.p2p          .tcpnet              Waiting for incoming connections...     {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3", "P2PID": "8sh6XhQN8mHeoxGivPLzviSAohGTUfjEW36k5aVwoWbj"}
2021-03-04T13:58:46.791-0500    DEBUG   9979a.p2p               Starting node bootstrap {"node_id": "9979a5e204dc2dad8bb3eff49260ff50ff1a7e0e22a087d1d47ad758d2060ad3", "P2PID": "8sh6XhQN8mHeoxGivPLzviSAohGTUfjEW36k5aVwoWbj"}
2021-03-04T13:58:46.791-0500    INFO    00000.defaultLogger     starting new grpc server with 5 registered service(s)
2021-03-04T13:58:46.791-0500    INFO    00000.defaultLogger     subscribed to channel
2021-03-04T13:58:46.791-0500    INFO    00000.defaultLogger     App started.
2021-03-04T13:58:46.791-0500    INFO    00000.defaultLogger     starting new grpc server on :9092
...
2021-03-04T13:58:46.791-0500    INFO    00000.defaultLogger     old grpc API listening on port 9091
2021-03-04T13:58:46.791-0500    INFO    00000.defaultLogger     json API listening on port 9090
```

### PR DESCRIPTION
**Pull request by dshulyak**
## Motivation
Make it easier to identify the node.
Closes: https://github.com/spacemeshos/go-spacemesh/issues/2318

## Changes
Example log 
```
starting spacemesh	{"data-dir": "/home/dd/.spacemesh-data/1", "post-dir": "/home/dd/post/data", "hostname": "dd-MS-7C37"}
```

## Test Plan
Existing test suite

## TODO
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed

## DevOps Notes
- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [ ] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by dshulyak** [comment]
@lrettig take a look please. I extended logger with hostname. i think it would be enough to easily identify the machine. also included local IP from routing information, but I am not sure if this is useful for your request.

**Review by lrettig — CHANGES_REQUESTED** [review]
I don't think we need this in every log line. That will substantially increase the size of the log files. I think it's sufficient to print it once when the node starts up. @noamnelke what do you think?

**Comment by dshulyak** [comment]
i think it is enough to print it once as well, i just implemented what you suggested in https://github.com/spacemeshos/go-spacemesh/issues/2318#issuecomment-793083191

**Comment by lrettig** [comment]
Fair point :) Sorry about the miscommunication. Let's just print it once, for now.

(For the record, the possible issue with only printing something once, at startup, is that we don't retain logs indefinitely. Testnet logs in kibana vanish after a few days. So having it in every log line definitely makes debugging easier, but I don't think it's absolutely essential or worth the cost of bigger log files.)

File: cmd/node/node.go [hunk 1]
```diff
@@ -970,16 +968,21 @@ func (app *SpacemeshApp) startSyncer(ctx context.Context) {
...
 
+	hostname, err := os.Hostname()
+	if err != nil {
+		logger.With().Error("error reading hostname", log.Err(err))
+	}
 	logger.With().Info("starting spacemesh",
 		log.String("data-dir", app.Config.DataDir()),
-		log.String("post-dir", app.Config.POST.DataDir))
+		log.String("post-dir", app.Config.POST.DataDir),
+		log.String("hostname", hostname),
```
Comment (lrettig) [inline cmd/node/node.go:972]: What does `hostname` print if there is an error? Assuming it's nil or an empty string, maybe set it to "(unknown)" or something instead to make it explicit?

Comment (dshulyak) [reply cmd/node/node.go:972]: sorry, i meant to exit here and didn't notice that it was just an error log (as on the line below).
i don't think we need to workaround a potential error. on Linux gethostname doesn't require special permissions, on other OS it probably the same. so if it fails it is either a flake, or something is wrong with env. so instead, I will just exit with an error.

Comment (dshulyak) [reply cmd/node/node.go:972]: also created a change for other similar cases https://github.com/spacemeshos/go-spacemesh/pull/2488

Comment (dshulyak) [reply cmd/node/node.go:972]: @lrettig i changed Fatal to Panic to preserve original behavior in this file until https://github.com/spacemeshos/go-spacemesh/pull/2488 is merged

**Review by lrettig — APPROVED** [review]

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**